core/src/main/clojure/xtdb/operator/scan.clj

```diff
@@ -608,6 +648,15 @@
                                               (expr/->expression-relation-selector select-form {:col-types col-types, :param-types param-types})))
                            (into {}))
 
+            iid-bb (when-let [eid-select (or (get selects 'xt/id) (get selects 'xt$id))]
```

Comment:
if you could split this out into a function, you could perhaps test that individually rather than relying on the `with-redefs`?